### PR TITLE
CCDE-1477: Add AWS Iceberg Converter

### PR DESCRIPTION
**Pull request by gwionap**
Added two converters:

- `AthenaIcebergSqlConverter`: Converter for taking mojap metadata and using it to create the necessary SQL queries for creating and / or updating an iceberg table in AWS using Athena.
- `AwsIcebergTable`: Converter for taking mojap metadata and using it to create / update an iceberg table using Athena. Can also take information from the AWS Glue Catalog and the Iceberg table's own metadata and convert this into mojap metadata.

This PR also includes adding a pre-commit config for running linting checks / unit tests locally prior to committing.

### CODE REVIEW

**Review by Thomas-Hirsch — COMMENTED** [review]
a couple of comments. I don't have enough of an understanding of iceberg to identify issue with the implementation, but seems generally sensible from an initial scan.

**Comment by Thomas-Hirsch** [comment]
I neglected to mention, but you'll also need to update the readme to explain how to use it.

File: mojap_metadata/converters/aws_iceberg_converter/__init__.py [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+import re
+from typing import Dict, List, Optional, Tuple, Union
+
+import awswrangler as wr
+import boto3
+from boto3.session import Session
+from dataengineeringutils3.s3 import read_json_from_s3
+from jinja2 import Template
+
+from ...metadata.metadata import Metadata
```
[the remark this answers is not in the document]
Comment (gwionap) [reply mojap_metadata/converters/aws_iceberg_converter/__init__.py]: Updated in latest commit

File: tests/test_aws_iceberg_converter.py [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+@pytest.fixture(scope="module")
+def sql_converter():
+    yield AthenaIcebergSqlConverter()
+
+
+@pytest.fixture(scope="module")
+def iceberg_metadata_dictionary():
+    with open("tests/data/aws_iceberg_converter/metadata.json", "r") as f:
+        meta_dict = json.load(f)
+    yield meta_dict
```
[the remark this answers is not in the document]
Comment (gwionap) [reply tests/test_aws_iceberg_converter.py]: Updated in latest commit

**Comment by gwionap** [comment]
@Thomas-Hirsch - do you think we could pick that up in a separate docs PR, that also maybe rejigs things to use mkdocs?

File: tests/test_aws_iceberg_converter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gwionap) [reply tests/test_aws_iceberg_converter.py]: Actually, had to move the `sql_converter` fixture back as `awswrangler` isn't a required dependency for the postgres converter.

**Review by Thomas-Hirsch — APPROVED** [review]

**Pull request merged**